components/datadog/fakeintake/docker.go

```diff
@@ -0,0 +1,52 @@
+package fakeintake
+
+import (
+	"log"
+	"net"
+
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/DataDog/test-infra-definitions/components"
+	"github.com/pulumi/pulumi-docker/sdk/v4/go/docker"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+const hostPort = 30080
+
+func NewLocalDockerFakeintake(e config.CommonEnvironment, resourceName string) (*Fakeintake, error) {
+	return components.NewComponent(e, resourceName, func(comp *Fakeintake) error {
+
+		_, err := docker.NewContainer(e.Ctx, e.CommonNamer.ResourceName("local-docker-container"), &docker.ContainerArgs{
+			Image: pulumi.String("public.ecr.aws/datadog/fakeintake:latest"),
+			Ports: docker.ContainerPortArray{
+				&docker.ContainerPortArgs{
+					Internal: pulumi.Int(80),
+					External: pulumi.Int(hostPort),
+				},
+			},
+		}, e.WithProviders(config.ProviderDocker))
+		if err != nil {
+			return err
+		}
+
+		localIP := getLocalIP()
+
+		comp.Host = pulumi.Sprintf(localIP.String())
+		comp.Port = hostPort
+		comp.Scheme = "http"
+		comp.URL = pulumi.Sprintf("%s://%s:%d", comp.Scheme, comp.Host, comp.Port)
+
+		return nil
+	})
+}
+
+func getLocalIP() net.IP {
+	conn, err := net.Dial("udp", "255.255.255.255:80") // initiate a connection to get the local address, the url does not need to exist
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
You should dial up on `hostPort` rather than on port `80`. If I understand it correctly, you are mapping the fakeintake container port 80 to the host port `30080`

Then we use it to connect to the fakeintake component URL from the test runner, that is the host where the fakeintake container instance is running.

I think we ca safely return `127.0.0.1:30080`

Reply:
Actually that's not needed. This one is just a trick to get your local private IP. This is explained probably better [here](https://gosamples.dev/local-ip-address/)
But the idea is to initiate a connection (we do not even care that the connection can actually be created). Once the connection is initiated you can retrieve the local IP from the connection object.

Reply:
> I think we ca safely return 127.0.0.1:30080
I would not be able to access the fake intake from within the `kind` cluster with that IP

Reply:
```suggestion
	// Open a connection to an external valid URL to  
	// get the local address from the connection instance
	// The URL does not need to exist
	conn, err := net.Dial("udp", "255.255.255.255:80") 
```

Reply:
What is confusing is that `255.255. 255.255` is the broadcast address: `Dial` is sending a broadcast UDP request to all the network. In our private subnets there is probably no service on port `80`. This message is still reaching n hosts, depending on the subnet mask

I'd rather use a fake URL or `8.8.8.8:80`

Reply:
Wanted something that make it clear that we do not need the IP to be reachable. Using the broadcast address was maybe not a good idea. I will move forward with `8.8.8.8:80`.
Thanks!